### PR TITLE
[Bakport] Adding product from wishlist not adding to cart showing warning message.

### PR DESCRIPTION
**Pull request by niravkrish**
### Original Pull Request 
https://github.com/magento/magento2/pull/19653

### Description (*)

Adding product from wishlist not adding to cart showing warning message.
### Fixed Issues (if relevant)

1. magento/magento2#9155: Adding product from wishlist not adding to cart showing warning message.

### Manual testing scenarios (*)

1. Add a simple product from admin with Minimum Qty Allowed in Shopping Cart 10
2. Login in front end
3. Add product to wishlist
4. Click on wishlist.
5. Now showing allow qty as default in qty field.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by dmytro-ch** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, here is your [new Magento instance](https://pr-22536.instances.magento-community.engineering).
Admin access: [https://pr-22536.instances.magento-community.engineering/admin](https://pr-22536.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @niravkrish,
the current solution does not seem to be working.
The `\Magento\Wishlist\Block\Customer\Wishlist\Item\Column\Cart` cannot be instantiated due to the redundant constructor params, and namely `\Magento\Catalog\Model\Product\Image\UrlBuilder` which is not available in Magento 2.2.x version.
Please make sure the changes are working and compatible with Magento 2.2.x before creating a PR.

File: app/code/Magento/Wishlist/Block/Customer/Wishlist/Item/Column/Cart.php [hunk 1]
```diff
@@ -14,6 +16,27 @@
...
+     * @param \Magento\Catalog\Block\Product\Context $context
+     * @param \Magento\Framework\App\Http\Context $httpContext
+     * @param array $data
+     * @param \Magento\Catalog\Block\Product\View $productView
+     */
+    public function __construct(
+        \Magento\Catalog\Block\Product\Context $context,
+        \Magento\Framework\App\Http\Context $httpContext,
+        \Magento\Catalog\Block\Product\View $productView = null,
+        array $data = []
```
Comment (dmytro-ch) [inline app/code/Magento/Wishlist/Block/Customer/Wishlist/Item/Column/Cart.php]: Please keep the same order of parameters as for the parent class.

File: app/code/Magento/Wishlist/Block/Customer/Wishlist/Item/Column/Cart.php [hunk 2]
```diff
@@ -14,6 +16,27 @@
  */
 class Cart extends \Magento\Wishlist\Block\Customer\Wishlist\Item\Column
 {
+    /**
+     * @var View
+     */
+    private $productView;
```
Comment (dmytro-ch) [inline app/code/Magento/Wishlist/Block/Customer/Wishlist/Item/Column/Cart.php:20]: Please add space between property and method.

File: app/code/Magento/Wishlist/Block/Customer/Wishlist/Item/Column/Cart.php [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 namespace Magento\Wishlist\Block\Customer\Wishlist\Item\Column;
 
+use Magento\Catalog\Controller\Adminhtml\Product\Initialization\StockDataFilter;
```
Comment (dmytro-ch) [inline app/code/Magento/Wishlist/Block/Customer/Wishlist/Item/Column/Cart.php]: This import does not seem to be used. Please remove.

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
@niravkrish, could you please squash all commits into a single commit after applying the changes in order to clean up history?
Thank you!

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-4859](https://jira.corp.magento.com/browse/ENGCOM-4859) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Comment by Rrego6** [comment]
Hi @niravkrish . Thank you for your contribution.  Unfortunately, we needed to revert this fix for the 2.2.9 release, as it includes BIC into a patch release.

**Pull request merged**